hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/strategy/CompactionStrategy.java

```diff
@@ -71,10 +76,14 @@ public Map<String, Double> captureMetrics(HoodieWriteConfig writeConfig, FileSli
    * @return Compaction plan to be scheduled.
    */
   public HoodieCompactionPlan generateCompactionPlan(HoodieWriteConfig writeConfig,
-      List<HoodieCompactionOperation> operations, List<HoodieCompactionPlan> pendingCompactionPlans) {
+      List<HoodieCompactionOperation> operations, List<HoodieCompactionPlan> pendingCompactionPlans, Map<String, String> params, Pair<List<String>, List<String>> partitionPair) {
     // Strategy implementation can overload this method to set specific compactor-id
+    Set<String> missingPartitions = new HashSet<>(partitionPair.getRight());
+    List<HoodieCompactionOperation> operationsToProcess = orderAndFilter(writeConfig, operations, pendingCompactionPlans, missingPartitions);
+    List<String> res = writeConfig.isIncrementalTableServiceEnable() ? new ArrayList<>(missingPartitions) : new ArrayList<>();
```

Comment:
Maybe null as the default value to avoid deserialization.

Reply:
chenged.